### ISSUE TITLE
DeepLearning-NLP-CV

### ISSUE THREAD

**Issue by Dharmin09** [Dharmin09/AI-Text-Summarization-Digit-Recognition#1]
# 🧠 AI-Based Projects: Text Summarization & Handwritten Digit Recognition 🚀  

This repository contains two deep learning projects:  

1️⃣ **Text Summarization using LSTM** (Natural Language Processing)  
2️⃣ **Handwritten Digit Recognition using CNN** (Computer Vision)  

Both projects are implemented using **TensorFlow/Keras** and trained on sample datasets.  

---

### 📝 **Overview**  
This project focuses on **text summarization** using **Long Short-Term Memory (LSTM) networks**. The model is trained on sample AI-related text data and uses **tokenization and padding** to prepare input sequences.  

### 🚀 **Features**  
✅ Implements **Natural Language Processing (NLP)** techniques.  
✅ Uses **LSTM (Long Short-Term Memory)** networks.  
✅ Tokenizes input text and converts it into numerical sequences.  

### 🔧 **Installation**  
Make sure you have Python installed and run:  
```sh
pip install tensorflow numpy
```

### 📛 **How to Run**  
Run the script using:  
```sh
python text_summarization_lstm.py
```

### 🖼️ **Sample Output**  
After training, the model prints:  
```
Text summarization model training complete!
```

---

### 📝 **Overview**  
This project implements a **Convolutional Neural Network (CNN)** to classify handwritten digits from `0` to `9` using the **MNIST dataset**. The model achieves **high accuracy** and can predict digits from unseen test data.  

### 🚀 **Features**  
✅ Uses **Convolutional Neural Networks (CNNs)**.  
✅ Trained on the **MNIST dataset**.  
✅ Achieves **97%+ accuracy**.  

### 🔧 **Installation**  
Make sure you have Python installed and run:  
```sh
pip install tensorflow matplotlib
```

### 📛 **How to Run**  
Run the script using:  
```sh
python digit_recognition_cnn.py
```

### 📛 **Dataset**  
- The **MNIST dataset** contains **60,000 training images** and **10,000 testing images** of handwritten digits.  

### 🖼️ **Sample Output**  
After training, the model prints the accuracy:  
```
Test accuracy: [value]
```
It also visualizes a sample test image with the predicted label.  

Example:  
![MNIST Example](https://upload.wikimedia.org/wikipedia/commons/2/27/MnistExamples.png)  

---

## 📛 **Common Requirements for Both Projects**  
Install all dependencies using:  
```sh
pip install tensorflow numpy matplotlib
```

---

## 👤 **Author**  
**Dharmin Tank**  

📧 **Contact:**  
- **GitHub:** [Dharmin09](https://github.com/Dharmin09)  
- **LinkedIn:** [Dharmin Tank](https://www.linkedin.com/in/dharmin09/)
